pkg/kubelet/eviction/helpers.go

```diff
@@ -1243,6 +1252,23 @@ func evictionMessage(resourceToReclaim v1.ResourceName, pod *v1.Pod, stats stats
 	if !ok {
 		return
 	}
+
+	// Pod level resources will be included in eviction message, along container resources
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources) && resourcehelper.IsPodLevelResourcesSet(pod) && resourcehelper.IsSupportedPodLevelResource(resourceToReclaim) {
+		podRequest := v1resource.GetResourceRequestQuantity(pod, resourceToReclaim)
+
+		var podUsage *resource.Quantity
+
+		if resourceToReclaim == v1.ResourceMemory && podStats.Memory != nil {
+			podUsage = memoryUsage(podStats.Memory)
+		}
+
+		message += fmt.Sprintf(podMessageFmt, pod.Name, podUsage.String(), podRequest.String(), resourceToReclaim)
+
+		annotations[OffendingPodKey] = pod.Name
+		annotations[OffendingPodUsageKey] = podUsage.String()
```

Comment:
podUsage could be nil here. I think this should all move into the above condition?

Reply:
I agree, thank you.